cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/common/PipelineRegisterer.java

```diff
@@ -94,9 +106,18 @@ public Pipeline registerPlugins(ETLConfig config) {
         throw new IllegalArgumentException(String.format("No Plugin of type '%s' named '%s' was found",
                                                          Constants.Transform.PLUGINTYPE, transformConfig.getName()));
       }
+      // if the transformation is configured to write filtered records to error dataset, we create that dataset.
+      if (transformConfig.getDatasetName() != null) {
+        // TODO : can remove this after implementing CDAP-3480
+        if (errorDatasetType != null) {
+          configurer.createDataset(transformConfig.getDatasetName(), errorDatasetType, DatasetProperties.builder()
```

Comment:
isnt the schema same between both of them? as we are writing InvalidEntry in both cases?